modules/scylladb/scylladb.go

```diff
@@ -0,0 +1,214 @@
+type Container struct {
+	testcontainers.Container
+}
+
+// WithConfig sets the YAML config file as an [io.Reader] to be used for the ScyllaDB container
+func WithConfig(r io.Reader) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			Reader:            r,
+			ContainerFilePath: "/etc/scylla/scylla.yaml",
+			FileMode:          0o644,
+		}
+		req.Files = append(req.Files, cf)
+
+		return nil
+	}
+}
+
+// WithShardAwareness enable shard-awareness in the ScyllaDB container so you can use the `19042` port.
+func WithShardAwareness() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.ExposedPorts = append(req.ExposedPorts, shardAwarePort)
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForListeningPort(shardAwarePort))
+		return nil
+	}
+}
+
+// WithAlternator enables the Alternator (DynamoDB Compatible API) service in the ScyllaDB container.
+// It will set the "alternator-port" parameter to the specified port.
+// It will also set the "alternator-write-isolation" parameter to "always" as a command line argument to the container.
+func WithAlternator(alternatorPort uint16) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		alternatorPortStr := strconv.FormatInt(int64(alternatorPort), 10)
+		req.ExposedPorts = append(req.ExposedPorts, alternatorPortStr)
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForListeningPort(nat.Port(alternatorPortStr)))
+		setCommandFlag(req, map[string]string{
+			"--alternator-port":            alternatorPortStr,
+			"--alternator-write-isolation": "always",
+		})
+
+		return nil
+	}
+}
+
+// WithCustomCommands sets custom commands with values for the ScyllaDB container.
+// Pass the command and the values as a list of strings in the following format: "--flag1=value", "--flag2", etc.
+// In case of an invalid flag (not starting with "--" or "-"), this option returns an error,
+// not applying any changes to the command line. Else, flags that exist in the command line overwrite the default commands.
+// See more in the [ScyllaDB docs].
+//
+// [ScyllaDB docs]: https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/tips/best-practices-scylla-on-docker.html
+func WithCustomCommands(flags ...string) testcontainers.CustomizeRequestOption {
+	var errInvalidFlag error
+
+	flagsMap := make(map[string]string)
+	for _, flag := range flags {
+		if !strings.HasPrefix(flag, "--") && !strings.HasPrefix(flag, "-") {
+			errInvalidFlag = fmt.Errorf("invalid flag: %s", flag)
+			break
+		}
+
+		before, after, found := strings.Cut(flag, "=")
+		if found {
+			flagsMap[before] = after
+		} else {
+			flagsMap[flag] = ""
+		}
+	}
+
+	return func(req *testcontainers.GenericContainerRequest) error {
+		if errInvalidFlag != nil {
+			return errInvalidFlag
+		}
```

Comment:
suggestion: merge the two blocks to simplify the logic
```suggestion
	return func(req *testcontainers.GenericContainerRequest) error {
		flagsMap := make(map[string]string, len(flags))
		for _, flag := range flags {
			if !strings.HasPrefix(flag, "--") && !strings.HasPrefix(flag, "-") {
				return fmt.Errorf("invalid flag: %s", flag)
			}

			before, after, found := strings.Cut(flag, "=")
			if found {
				flagsMap[before] = after
			} else {
				flagsMap[flag] = ""
			}
		}
```